### ISSUE TITLE
Improve exports

### ISSUE THREAD

**Issue by ueman** [cfug/dio#1362]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          | all  |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              | all  |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  | 4.0.4|
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              | any  |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   | 100% |
| Repro with our demo prj        | e.g. does it happen with our demo project?            | no   |
| Demo project link              | e.g. link to a demo project that highlights the issue | mo   |

### Issue Description and Steps

I would like to have a cross environment `DefaultHttpClientAdapter`. At the moment you have to import `import 'package:dio/adapter_browser.dart';` or `import 'package:dio/adapter.dart';` depending on whether you are using `dart:html` or `dart:io`.

There are also a couple exports which are basically duplicated for `dart:html` and `dart:io`.
There's `browser_imp.dart` and `adapter_browser.dart` for web and `adapter.dart` and `native_imp.dart` for dart:io.

I suggest changing this to the following:

- `package:dio/universal_adapter.dart` -> exposes a adapter which creates an adapter based on the underlying platform (e.g. web adapter for web and io adapter for io)
- `package:dio/html.dart` - a single export for all web stuff
- `package:dio/io.dart` - a single export for all io stuff
- deprecate the other exports and remove them in a future version

If you want, I can create a PR to further demonstrate or implement what I mean.

**Comment by kuhnroyal** [comment]
I agree, the imports can be improved and some sort of `PlatformHttpClientAdapter` can be added.
But the thing is, both current adapters allow for completely different ways of customization and if you want to customize, you need to know what adapter it is and need to use conditional imports.
